### ISSUE TITLE
[tvOS] Add-on couldn't be loaded. An unknown error has occurred.

### ISSUE THREAD

**Issue by NicoxDJ** [xbmc/inputstream.ffmpegdirect#196]
I tried to install this plugin on my tvOS.
Apparently it isn't working :(
I did manage to install the plugin and even I can check it's settings, but when I try to start a channel, it shows me the error from the title.

Kodi Version: 19.4
tvOS Version: 15.6

I installed Kodi via this way: https://kodi.wiki/view/HOW-TO:Install_Kodi_on_Apple_TV_4_and_5_(HD_and_4K)
I downloaded the plugin from here: https://jenkins.kodi.tv/job/xbmc/job/inputstream.ffmpegdirect/job/Matrix/108/
I tried even 19.0.1, because I read some comments that it works, but it doesn't for me too.

What I'm doing wrong? How can I successfully install that plugin?

**Comment by phunkyfish** [comment]
It should be available as default when you install kodi on tvos. TVOS is different in that the addons are packaged with kodi. So you just need to enable the addons.

**Comment by NicoxDJ** [comment]
@phunkyfish well, it wasn't in my plug-ins, and I can't find it neither in the repo.
So that's why I google it and found it here

**Comment by phunkyfish** [comment]
And the other inputstreams were in the plug-ins just not this one?

The reason why it’s packaged with kodi is that it must be signed with the package that is being installed.

**Comment by NicoxDJ** [comment]
@phunkyfish that's right, the other 2 were there (InputStream Adaptive and RTMP Input). Just this was missing.

How I can sign it? Is somewhere written procedure? Or I should try to download some older version of Kodi and try it?

**Comment by phunkyfish** [comment]
We need to figure out why it didn’t build as part of 19.4. I don’t think you can build and sign it afterwards. That won’t work.

**Comment by phunkyfish** [comment]
@sy6sy2 any guidance to figure out what the build issue is?

**Comment by NicoxDJ** [comment]
I just unpack the deb file, and here is the list with all addons in the package:
https://pastebin.com/Rvgx5v1P

**Comment by NicoxDJ** [comment]
@phunkyfish FYI, I downloaded and checked all 19.* versions. None of them have the plugin inside it.
I think I have to drag the addon inside and pack the data.tar.gz, then pack the .deb app too, and after that I have to sign in with the 'iOS App Signer' ?

**Comment by phunkyfish** [comment]
I think you should try that but I’m not sure if it will work. @kambala-decapitator do you happen to know if adding a binary addon in this way and resigning would work? Or would we have to build it all as part of an official release/build?

**Comment by kambala-decapitator** [comment]
> I think I have to drag the addon inside and pack the data.tar.gz, then pack the .deb app too, and after that I have to sign in with the 'iOS App Signer' ?

yes, you're right. But you don't need to pack to deb again - you can sign the .app with App Signer ;)

unpack deb:

```bash
ar -p YOUR_DEB_FILE data.tar.xz | tar -xf -
```

New directory Applications will appear, inside there will be Kodi.app - place the add-on in `Kodi.app/AppData/AppHome/addons`. After that drag Kodi.app into App Signer.

**Comment by NicoxDJ** [comment]
@phunkyfish @kambala-decapitator yes, it worked this way. Thank you!

**Comment by phunkyfish** [comment]
Great!

**Comment by socksprox** [comment]
@phunkyfish Do you still plan on fixing it? It completely breaks iptv on Kodi for me. I think this has been kinda forgotten?

**Comment by phunkyfish** [comment]
> @phunkyfish Do you still plan on fixing it? It completely breaks iptv on Kodi for me. I think this has been kinda forgotten?

Do you need to post on every issue? It will get fixed when the author of the required PR gets time to fix it. We are all volunteers here and sometimes you just need to wait.

**Comment by socksprox** [comment]
> > @phunkyfish Do you still plan on fixing it? It completely breaks iptv on Kodi for me. I think this has been kinda forgotten?
> 
> Do you need to post on every issue? It will get fixed when the author of the required PR gets time to fix it. We are all volunteers here and sometimes you just need to wait.

Sorry, you’re right, i just thought this has been forgotten.

**Comment by phunkyfish** [comment]
No worries, thanks for being understanding.
